frontend/src/components/TruncatedText.tsx

```diff
@@ -6,21 +6,47 @@ type TruncatedTextProps = {
   content: string;
 } & React.HTMLProps<HTMLSpanElement>;
 
-const TruncatedText: React.FC<TruncatedTextProps> = ({ maxLines, content, ...rest }) => (
-  <Tooltip content={content}>
+const TruncatedText: React.FC<TruncatedTextProps> = ({ maxLines, content, ...props }) => {
+  const outerElementRef = React.useRef<HTMLElement>(null);
+  const textElementRef = React.useRef<HTMLElement>(null);
+  const [isTruncated, setIsTruncated] = React.useState<boolean>(false);
+
+  const updateTruncation = React.useCallback(() => {
+    if (textElementRef.current && outerElementRef.current) {
+      setIsTruncated(textElementRef.current.offsetHeight > outerElementRef.current.offsetHeight);
+    }
+  }, []);
+
+  const truncateBody = (
     <span
       style={{
         display: '-webkit-box',
-        overflow: 'hidden',
         WebkitBoxOrient: 'vertical',
-        WebkitLineClamp: maxLines,
         overflowWrap: 'anywhere',
+        overflow: 'hidden',
+        WebkitLineClamp: maxLines,
+      }}
+      ref={outerElementRef}
+      onMouseEnter={(e) => {
+        props.onMouseEnter?.(e);
+        updateTruncation();
+      }}
+      onFocus={(e) => {
+        props.onFocus?.(e);
+        updateTruncation();
       }}
-      {...rest}
     >
-      {content}
+      <span ref={textElementRef} {...props}>
```

Comment:
I think `props` should be applied to the outer span. For instance if someone wanted to apply margins or borders, it should be applied to the outer because the inner one is being truncated and not fully visible.

Spread the `props` before `style` and also spread `props.style` into the style before any of the fixed styles.

Reply:
I spread the style props after the fixed styles in case, for some reason, someone wants to override one of them. Maybe there would be a reason.